File: 1254-DeepestLeavesSum/1254-DeepestLeavesSum.java
// Last updated: 3/26/2026, 1:50:16 AM
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    
    public int deepestLeavesSum(TreeNode root) {
        if(root == null)
            return 0;
        int height =  getHeight(root);
        
        Queue<TreeNode> queue = new LinkedList<>();
        queue.add(root);
        int sum = 0;
        int level = 0;
        while(!queue.isEmpty()){
            level++;
            int size = queue.size();
            
            for(int i=0;i<size;i++){
                TreeNode temp = queue.poll();
                if(level==height)
                    sum+=temp.val;
                if(temp.left!=null)
                    queue.add(temp.left);
                if(temp.right!=null)
                    queue.add(temp.right);
            }
        }
        
        
        return sum;
    }
    
    private int getHeight(TreeNode root){
        if(root == null)
            return 0;
        int left = getHeight(root.left);
        int right = getHeight(root.right);
        
        return Math.max(left,right)+1;
    }
}

File: 1254-DeepestLeavesSum/1254-DeepestLeavesSum.cs
// Last updated: 3/26/2026, 1:50:18 AM
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public int DeepestLeavesSum(TreeNode root) {
        var sum = 0;
        var queue = new Queue<TreeNode>();
        queue.Enqueue(root);
        while(queue.Count > 0) {
            sum = 0;
            var count = queue.Count;
            for(var i = 0; i< count; i++) {
                var node = queue.Dequeue();
                sum += node.val;
                if (node.left != null) queue.Enqueue(node.left);
                if (node.right != null) queue.Enqueue(node.right);
            }
        }
        return sum;
    }
}